### DISCUSSION TITLE
Modem shell with modem_cellular

### DISCUSSION THREAD

**Discussion by johnp789** [zephyrproject-rtos/zephyr#65483] [Q&A]
I'm working on an application similar to the `cellular-modem` sample.  My board has a Quectel BG95 modem.  The application is able to establish a connection and ping internet hosts.  Now, I'd like to add `CONFIG_MODEM_SHELL=y` so that I can get modem information like the IMEI and send AT commands.  Adding `CONFIG_MODEM_CONTEXT=y` gets the modem shell command compiled into my application, but the `modem list` command does not return any modems.  

Is the new `modem_cellular` driver interoperable with the modem shell commands?  If so, what might I be missing in order to get the modem shell list to find the modem?

**Comment by ouaaliw** [comment]
Hello Mr.John, I am in the same trouble, i could not fix this problem, do you have please any idea to fix this problem?

**Reply by johnp789** [reply]
I did not find a solution, but I did not spend much time on it. In the end, I used something other than Zephyr for my project, for non-technical reasons.

**Comment by erickmf** [comment]
Same problem here with the SIM7080. Modem shell is enabled, but modem list doesn't show anything.
I'm using Zephyr LTS v3.7.1.

Any tips?

**Comment by zafersn** [comment]
Hey @johnp789,

I am also working with `modem_cellular` sample over at `zephyr/samples/net/cellular_modem`. To send AT commands with this setup, you just need to enable `CONFIG_MODEM_AT_SHELL`.

Not quite sure but It looks like you might've turned off `CONFIG_MODEM_CELLULAR` and switched on `CONFIG_MODEM_QUECTEL_BG9X`. That means you're using the dedicated Quectel BG9X driver, which might not play nice with the `modem_cellular` setup.

I am trying to figure out on your current setup and which modules you're using? It seems like you're working with an older modem driver and its related shell and context. I haven't tested those with `modem_cellular`, so I'm not sure how well they mesh.

If you're able to send AT commands through the shell but the modem list isn't showing up, you might want to check out these functions in modem_shell.c:

- `cmd_modem_list`
- `ms_context_from_id`
- `modem_context_from_id`

These are key in pulling up the modem context and might shed some light on what's going on.
